dll/win32/browseui/addresseditbox.cpp

```diff
@@ -104,6 +104,86 @@ BOOL CAddressEditBox::GetComboBoxText(CComHeapPtr<WCHAR>& pszText)
     return fCombobox.GetWindowText(pszText, cchMax);
 }
 
+HRESULT CAddressEditBox::RefreshAddress()
+{
+    CComPtr<IShellFolder> sf;
+    HRESULT hr;
+    CComHeapPtr<ITEMIDLIST> absolutePIDL;
+    LPCITEMIDLIST pidlChild;
+    STRRET ret;
+    WCHAR szName[4096], szPath[MAX_PATH];
+
+    if (pidlLastParsed)
+        ILFree(pidlLastParsed);
+    pidlLastParsed = NULL;
+
+    /* Get the current pidl of the browser */
+    hr = GetAbsolutePidl(&absolutePIDL);
+    if (FAILED(hr))
+        return hr;
+
+    if (!absolutePIDL)
+    {
+        ERR("Got no PIDL, investigate me!\n");
+        return S_OK;
+    }
+
+    /* Fill the combobox */
+    PopulateComboBox(absolutePIDL);
+
+    /* Find the current item in the combobox and select it */
+    CComPtr<IShellFolder> psfDesktop;
+    hr = SHGetDesktopFolder(&psfDesktop);
+    if (FAILED_UNEXPECTEDLY(hr))
+        return S_OK;
+
+    hr = psfDesktop->GetDisplayNameOf(absolutePIDL, SHGDN_FORADDRESSBAR, &ret);
+    if (FAILED_UNEXPECTEDLY(hr))
+        return S_OK;
+
+    hr = StrRetToBufW(&ret, absolutePIDL, szName, 4095);
```

Comment:
Why not using `_countof(szName)` there? (also why removing one character here, but not at the other StrRetToBufW call below?)

Reply:
It's better to ask these questions to the original code author?